### PR TITLE
Reorganize Ed25519 implementation

### PR DESCRIPTION
**Pull request by alinush**
### Description

Just moving files around for now. 

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2225)

### CODE REVIEW

File: crates/aptos-crypto/src/lib.rs [hunk 1]
```diff
@@ -24,6 +24,11 @@ mod unit_tests;
 pub use self::traits::*;
 pub use hash::HashValue;
 
+// We need to add this here if we want to use aptos-crypto-derive's CryptoHasher and BCSCryptoHasher
```
Comment (alinush) [inline crates/aptos-crypto/src/lib.rs:27]: I will fix this later.

The problem (I think) lies in https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos-crypto-derive/src/lib.rs#L398 and in https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos-crypto-derive/src/lib.rs#L428, which generate a `use aptos_crypto::...` line that will not work when generated inside the `aptos-crypto` crate itself (e.g., like in our Ed25519 PoK implementation in `crates/aptos-crypto/src/ed25519/ed25519_pok.rs`).

File: aptos-move/framework/src/natives/signature.rs [hunk 2]
```diff
@@ -123,6 +123,46 @@ pub fn native_ed25519_signature_verification(
...
+    context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut arguments: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(_ty_args.is_empty());
+    debug_assert!(arguments.len() == 2);
+
+    // Verifying a proof-of-knowledge is more-or-less an Ed25519 signature verification, with a litte
+    // extra serialization cost.
+    let msg_len = ed25519::ED25519_PUBLIC_KEY_LENGTH * 2;
```
Comment (wrwg) [inline aptos-move/framework/src/natives/signature.rs:136]: Add a TODO here to tackle this with the upcoming gas schedule changes. We will have a special formula for this function.

**Review by wrwg — COMMENTED** [review]
LGTM from Move side. Please approve someone with better knowledge from the crypto end.

**Comment by zekun000** [comment]
I don't understand what's the use case of this, can someone remind me of that?

**Review by zekun000 — APPROVED** [review]

File: crates/aptos-crypto/src/ed25519/mod.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+];
+
+pub mod ed25519_keys;
+pub mod ed25519_sigs;
+
+#[cfg(any(test, feature = "fuzzing"))]
+pub use ed25519_keys::keypair_strategy;
+
+pub use ed25519_keys::{Ed25519PrivateKey, Ed25519PublicKey};
+pub use ed25519_sigs::Ed25519Signature;
```
Comment (davidiw) [inline crates/aptos-crypto/src/ed25519/mod.rs:53]: can we actually improt these also as 
ed25519::private_key, ed25519::public_key, and ed25519::signature

which will be consistent with bls and x25519

also not sure why we split the files up 🤷‍♂️

Comment (alinush) [reply crates/aptos-crypto/src/ed25519/mod.rs:53]: **1.** The reason this PR is just splitting files is I was adding extra proof-of-knowledge (PoK) functionality (which I removed yesterday due to unclear use-case) and I wanted to structure this similar to the `bls12381/` directory. Also, the single `ed25519.rs` file was rather big.

**2.** Sure, I assume you mean capitalized?

In BLS, we have:

```
pub use bls12381_keys::{PrivateKey, PublicKey};
pub use bls12381_pop::ProofOfPossession;
pub use bls12381_sigs::Signature;
```

In x25519, we also export `x25519::PublicKey` and `x25519::PrivateKey`.

Comment (davidiw) [reply crates/aptos-crypto/src/ed25519/mod.rs:53]: Woops capitalized indeed! Thanks for the explanation!

**Pull request merged**